azerothcore/azerothcore-wotlk
# fix(Scripts/Midsummer): Implement Ribbon Pole dance effects

## Changes Proposed:
This PR proposes changes to:

If your pull request promotes complex changes that require a detailed explanation, please describe them in detail specifying what your solution is and what is it meant to address.

## Issues Addressed:

- Closes #12145
- Closes chromiecraft/chromiecraft#3647

## SOURCE:

The changes have been validated through:

## Tests Performed:

This PR has been:


## How to Test the Changes:



.event start 1
.tele orgrimmar
.go xyz 1937 -4326 22
click on ribbon pole

observe effects depending on dancing player count:

1 random blue / red ring of fire
-> 50% chance, updates every 5s

2 add fireworks (1 rocket)
-> 15s interval

3 add ground flowers
-> 10s interval, max. 3

3 upgrade fireworks to 3 rockets

5 upgrade fireworks to 5 rockets

6 makes it spew lava
-> choosing 2 random targets every 10 seconds

7 people make big dancing flames spawn
-> 30s interval, max. 4, despawn after 5min

If a dancing player leaves the pole, the effects get reduced to the previous level.

The thresholds, timings and max. counts of spawned NPCs can be modified in the enum RibbonPole, e.g. setting all THRESHOLD_...'s to 1 will trigger all effects for 1 dancing player for testing purposes.

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
